spark/src/main/scala/geotrellis/spark/io/hadoop/geotiff/GeoTiffLayerReader.scala

```diff
@@ -0,0 +1,82 @@
+package geotrellis.spark.io.hadoop.geotiff
+
+import geotrellis.raster.io.geotiff._
+import geotrellis.raster.{Raster, RasterExtent, Tile}
+import geotrellis.raster.resample.ResampleMethod
+import geotrellis.spark.tiling.ZoomedLayoutScheme
+import geotrellis.spark.{LayerId, SpatialKey}
+import geotrellis.vector.{Extent, ProjectedExtent}
+import geotrellis.raster.crop.Crop
+import geotrellis.raster.reproject.Reproject.{Options => ReprojectOptions}
+
+import java.net.URI
+
+// global context only for test purposes, should be refactored
+import scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
Generally not safe, to rely on global thread pool for IO.
Ideally we could just capture a `Monoid` that would either be a `Future[V]`, an `Option[V]` or `Future[Option[V]]` but I'm not certain how that would discover the thread pool though.

Reply:
Let's make it an implicit context, I agree. 
But in the future that should be an own blocking threadpool